application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,351 @@
+                logger.debug(
+                        "Channel status is currently busy, ignoring message received in {} from {}",
+                        guildEvent.getChannel().getName(), guildEvent.getAuthor());
+                return;
+            }
+            channelMonitor.setChannelBusy(guildEvent.getChannel().getIdLong(),
+                    guildEvent.getAuthor().getIdLong());
+            displayStatus(channelMonitor.getStatusChannelFor(guildEvent.getGuild()));
+            guildEvent.getMessage().reply(UserStrings.NEW_QUESTION.message()).queue();
+        }
+    }
+
+    private @NotNull Optional<Message> getStatusMessageIn(@NotNull TextChannel channel) {
+        if (!channelIdToMessageIdForStatus.containsKey(channel.getIdLong())) {
+            return findExistingStatusMessage(channel);
+        }
+        return Optional.ofNullable(channelIdToMessageIdForStatus.get(channel.getIdLong()))
+            .map(channel::retrieveMessageById)
+            .map(RestAction::complete);
+    }
+
+    private @NotNull Optional<Message> findExistingStatusMessage(@NotNull TextChannel channel) {
+        // will only run when bot starts, afterwards its stored in a map
+
+        Optional<Message> statusMessage = FreeUtil
+            .getChannelHistory(channel, FreeCommandConfig.MESSAGE_RETRIEVE_LIMIT)
+            .flatMap(history -> history.stream()
+                .filter(message -> !message.getEmbeds().isEmpty())
+                .filter(message -> message.getAuthor().equals(channel.getJDA().getSelfUser()))
+                // FIXME the equals is not working, i believe its because there is no getTitleRaw()
+                // .filter(message -> STATUS_TITLE.equals(message.getEmbeds().get(0).getTitle()))
+                .findFirst());
+
+        channelIdToMessageIdForStatus.put(channel.getIdLong(),
+                statusMessage.map(Message::getIdLong).orElse(null));
+        return statusMessage;
+    }
+
+    private void initChannelsToMonitor() {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getMonitoredChannels)
+            .flatMap(Collection::stream)
+            .forEach(channelMonitor::addChannelToMonitor);
+    }
+
+    private void initStatusMessageChannels(@NotNull final JDA jda) {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getStatusChannel)
+            .map(jda::getTextChannelById)
+            // throws NPE if id from config does not match a text channel
+            .map(Objects::requireNonNull)
+            .forEach(channelMonitor::addChannelForStatus);
+    }
+
+    private void initStatusMessages(@NotNull final JDA jda) {
+        // Attempts to find the existing status message, for all guilds. On launch.
+        channelMonitor.statusIds()
+            .map(jda::getTextChannelById)
+            .filter(Objects::nonNull) // only here for sonarLint
```

Comment:
why? what did sonar say? and why do u think its irrelevant?

Reply:
object could be null

before the id's are added to the map
they get tested for being a valid id for the correct guild and that they are a textchannel before being added to the map
so any call to the jda using an id from the map will work (unless the channel gets deleted after boot and before the call)
at which point that call should crash with an NPE, so the warning is stupid

Reply:
okay, maybe add that as comment instead. like `// can not be null, ensured that at insert, but better safe than sorry`

Reply:
or, if you want to be fail fast, remove the filter and use `Objects.requireNonNull` in ur following action.

Reply:
as in "dont trust ur own code". i.e. what if the other method is changed, then this method might want to be able to capture that illegal setup fast.

Reply:
well, if we're catching the same problems everywhere it should be done properly
an NPE will not give any relevant info here

made custom method to throw IllegalState with a useful error message

Reply:
fixed